tests/Unit/Adapter/ToolsTest.php

```diff
@@ -58,4 +58,14 @@ public function provideTestCasesForBcAdd(): iterable
         yield ['0.0000000', '10', 6, '10.000000'];
         yield ['0.0', '0.00000002', 2, '0.00'];
     }
+
+    /**
+     * Test for refreshCaCertFile : delete de Ca Cert file, refresh and test if the file is created
+     */
+    public function testRefreshCaCertFile(): void
+    {
+        @unlink(_PS_CACHE_CA_CERT_FILE_);
+        (new Tools())->refreshCaCertFile();
```

Comment:
Hello @jf-viguier this is a good start! May I suggest some improvements ideas?

1. I'm a bit afraid if you delete the `_PS_CACHE_CA_CERT_FILE_` because if for some reasons `(new Tools())->refreshCaCertFile()` fails, then your test shop has now no `_PS_CACHE_CA_CERT_FILE_` file 😄 

So I suggest to _modify_ the value of _PS_CACHE_CA_CERT_FILE_ just before the test, to target another dummy file that _you control_. And if this dummy file is deleted, it's OK, it's not bad for the shop.

2. Your assertion is "the file exists? yes, the test is successfull !" but it means that even if the file contains unusable data like `<husakwqàihcàchzcaçàu` the test will be successfull. 

So I suggest to validate that the file exists _and_ to validate the content is what we expect 😄

Reply:
1. @matks : the cacert.pem created by the test is in ./var/cache/test/cacert.pem
The prod or dev cacert.pem are in another folder (./var/cache/dev or ./var/cache/prod)
So there is no risk at all.
2. I don't know how to test the validity of cacert.pem. It could be done by another test. This test is for file generation.
The file validity is not checked by prestashop nor ps_checkout so I guess the validity check is useless in the test and should be done in the core first.

For the record, if remote url is not ok (https://curl.haxx.se/ca/cacert.pem) the process uses a local copy in ./vendor/composer/ca-bundle/res/ 
So, the risk of having an empty or corrupt file is really low.

Reply:
Agree for the point 1 @jf-viguier 

For the second point, I agree with @matks :

```suggestion
        self::assertEquals(1, file_exists(_PS_CACHE_CA_CERT_FILE_));
        $original = @file_get_contents(Tools::CACERT_LOCATION, false, $stream_context);
            if (empty($original)) {
                $original = @file_get_contents(CaBundle::getBundledCaBundlePath());
            }
        self::assertEquals($original, file_get_contents(_PS_CACHE_CA_CERT_FILE_));
```

Reply:
@Progi1984 I've debugged and added your suggestion